lib/adb.js

```diff
@@ -1,1700 +1,66 @@
-  this.shell('ime list -a',  function (err, stdout) {
-    if (err) return cb(err);
-    var engines = [];
-    _.each(stdout.split('\n'), function (line) {
-      // get a listing that has IME IDs flush left,
-      // and lots of extraneous info indented
-      if (line.length > 0 && line[0] !== ' ') {
-        // remove newline and trailing colon, and add to the list
-        engines.push(line.trim().replace(/:$/, ''));
-      }
-    });
-    cb(null, engines);
-  });
-};
-
-ADB.prototype.defaultIME = function (cb) {
-  var cmd = 'settings get secure default_input_method';
-  this.shell(cmd, function (err, engine) {
-    if (err) return cb(err);
-    cb(null, engine.trim());
-  });
-};
-
-ADB.prototype.enableIME = function (imeId, cb) {
-  var cmd = 'ime enable ' + imeId;
-  this.shell(cmd, cb);
-};
-
-ADB.prototype.disableIME = function (imeId, cb) {
-  var cmd = 'ime disable ' + imeId;
-  this.shell(cmd, cb);
-};
-
-ADB.prototype.setIME = function (imeId, cb) {
-  var cmd = 'ime set ' + imeId;
-  this.shell(cmd, cb);
-};
-
-ADB.prototype.hasInternetPermissionFromManifest = function (localApk, cb) {
-  this.checkAaptPresent(function (err) {
-    if (err) return cb(err);
-    logger.debug("Checking if has internet permission from manifest.");
-    prettyExec(this.binaries.aapt,
-               ['dump', 'badging', localApk],
-               { maxBuffer: 524288 }, function (err, stdout, stderr) {
-      if (err || stderr) {
-        logger.warn(stderr);
-        return cb(new Error("hasInternetPermissionFromManifest failed. " + err));
-      }
-      var hasInternetPermission = new RegExp("uses-permission:.*'android.permission.INTERNET'").test(stdout);
-      cb(null, hasInternetPermission);
-    });
-  }.bind(this));
-};
-
-ADB.prototype.reboot = function (cb) {
-  var adbCmd = "stop; sleep 2; setprop sys.boot_completed 0; start";
-  this.shell(adbCmd, function (err) {
-    if (err) return cb(err);
-    var bootCompleted = false;
-    var i = 90;
-    logger.debug('waiting for reboot, this takes time.');
-    async.until(
-      function test() { return bootCompleted; },
-      function fn(cb) {
-        i--;
-        if (i < 0) return cb(new Error('device didn\'t reboot within 90 seconds'));
-        if (i % 5 === 0) logger.debug('still waiting for reboot.');
-        this.shell("getprop sys.boot_completed", function (err, stdout) {
-          if (err) return cb(err);
-          bootCompleted = '1' === stdout.trim();
-          setTimeout(cb, 1000);
-        });
-      }.bind(this),
-      cb
-    );
-  }.bind(this));
-};
-
-ADB.getAdbServerPort = function () {
-  return process.env.ANDROID_ADB_SERVER_PORT || 5037;
-};
-
-module.exports = ADB;
+// transpile:mocha
+
+import _ from 'lodash';
+import path from 'path';
+import methods from './tools/index.js';
+
+class ADB {
+
+ constructor (opts = {}) {
+   if (typeof opts.sdkRoot === "undefined") {
+     opts.sdkRoot = process.env.ANDROID_HOME || '';
+   }
+   const defaults = {
```

Comment:
Do we have a set style for these sorts of constants? I myself would prefer it to be in the root scope of the module, as something like `const DEFAULT_OPTS = {...}`, so it is visible as what it is. A constant in function scope seems... odd.